File: stack/1673_find_the_most_competitive_subsequence/1673_find_the_most_competitive_subsequence.go
package main

import "fmt"

func mostCompetitive(nums []int, k int) []int {
	res := []int{}
	k = len(nums) - k
	for i := range nums {
		for len(res) > 0 && k > 0 && res[len(res)-1] > nums[i] {
			res = res[:len(res)-1]
			k--
		}
		res = append(res, nums[i])
	}
	return res[:len(res)-k]
}

func main() {
	nums := []int{3, 5, 2, 6}
	k := 2
	res := mostCompetitive(nums, k)
	fmt.Println(res)
}

File: stack/1673_find_the_most_competitive_subsequence/1673_find_the_most_competitive_subsequence.py
from typing import List


class Solution:
    def mostCompetitive(self, nums: List[int], k: int) -> List[int]:
        ans = []
        r = len(nums) - k
        for i in range(len(nums)):
            while len(ans) > 0 and ans[-1] > nums[i] and r > 0:
                ans.pop()
                r -= 1
            ans.append(nums[i])
        return ans[:k]
        

nums = [3,5,2,6]
k = 2
res = Solution().mostCompetitive(nums, k)
print(res)